packages/react-components/src/shared/src/forwardRef.ts

```diff
@@ -40,6 +40,7 @@ type MergeWithAs<T, P> = RightJoinProps<PropsOf<T>, OmitCommonProps<P, "slot"> &
      * Default slot override. Added to every orbit component
      */
     slot?: string;
+    "data-testid"?: string;
```

Comment:
This will conflict with the Merge with As pr. 

`data-` are not automatically accepted by components. With typescript 4.4, there could be a workaround : 
https://github.com/microsoft/TypeScript/issues/28960#issuecomment-903519759

This is currently a limitation of typescript. We're waiting for microsoft/TypeScript#28960.
For more context see #18902.
Possible workarounds:
@ts-ignore:
<Checkbox
  inputProps={{
    // @ts-ignore
    'data-testid': 'test-checkbox',
  }}
/>;
use ByRole instead of ByTestid. We generally don't recommend data-testid. Most of the (interactions) tests can be implemented using ByRole